core/src/tile/mbtilesTileTask.cpp

```diff
@@ -0,0 +1,105 @@
+#include "mbtilesTileTask.h"
+
+#include "log.h"
+#include "data/dataSource.h"
+#include "hash-library/md5.cpp"
+
+namespace Tangram {
+
+MBTilesTileTask::MBTilesTileTask(TileID& _tileId, std::shared_ptr<DataSource> _source, int _subTask) :
+    DownloadTileTask(_tileId, _source, _subTask), m_db(_source->mbtilesDb()),
+    m_getTileDataStmt(m_db, "SELECT tile_data FROM tiles WHERE zoom_level = ? AND tile_column = ? AND tile_row = ?;"),
+    m_putMapStmt(m_db, "REPLACE INTO map (zoom_level, tile_column, tile_row, tile_id) VALUES (?, ?, ?, ?);"),
+    m_putImageStmt(m_db, "REPLACE INTO images (tile_id, tile_data) VALUES (?, ?);") {
+
+}
+
+/**
+ * An MBTilesTileTask is created when we have a path to an MBTiles file for the DataSource.
+ * Rather than assuming we have already gotten our rawData to parse, we check to see if
+ * we have rawData. If not, we query the SQLite MBTiles database for the rawData that we seek.
+ *
+ * @param _tileBuilder
+ */
+void MBTilesTileTask::process(TileBuilder &_tileBuilder) {
+    if (hasData()) {
+        // If the data did not come from the in-memory cache, it is new.
+        // It should be added to the MBTiles.
+        if (!dataFromCache) {
+            putMBTilesData();
```

Comment:
Generic question, how much of performance gain we get when we grab the date from cache, vs when we run the query to grab the tile data every time. Just debating on memory vs performance benefit here?